src/java/htsjdk/samtools/metrics/MetricBase.java

```diff
@@ -84,7 +85,10 @@ public int hashCode() {
         int result = 0;
         for (final Field f : getClass().getFields()) {
             try {
-                result = 31 * result + f.get(this).hashCode();
+                Object value = f.get(this);
+                value = new FormatUtil().format(value); //format the value the way it will be written to disk
```

Comment:
I agree, it doesn't seem ideal, but the alternative possibly more performant way was was pretty gross and brittle. 


Will do.  Any objection to either 
+ creating a default formatter singleton 

or

+  rewrite FormatUtil to just have static methods, possibly breaking any client code the extends FormatUtils.  (which would cleaner but maybe unpalatable)